src/mon/OSDMonitor.cc

```diff
@@ -5128,19 +5129,32 @@ bool OSDMonitor::handle_osd_timeouts(const utime_t &now,
     }
     if (!osdmap.is_up(i))
       continue;
-    const std::map<int,utime_t>::const_iterator t = last_osd_report.find(i);
+    const std::map<int, std::pair<utime_t, int>>::const_iterator t = last_osd_report.find(i);
     if (t == last_osd_report.end()) {
       // it wasn't in the map; start the timer.
-      last_osd_report[i] = now;
+      last_osd_report[i].first = now;
+      last_osd_report[i].second = g_conf()->osd_beacon_report_interval;
     } else if (can_mark_down(i)) {
-      utime_t diff = now - t->second;
+      utime_t diff = now - t->second.first;
       if (diff > timeo) {
```

Comment:
I think a better approach would be to set timeout here to something like
```
utime_t timeo(std::max(g_conf()->mon_osd_report_timeout, t->second.second *2))
```
so that we are implicitly using the larger timeout for the misconfigured OSD only... and only for the period(s) when the OSD is misconfigured.  (this also needs to handle the case when t->second.second is utime_t())

Reply:
do you mean that we just use the max one to checkout timeout and  don't need to change the mon_osd_report_timeout to 2*osd_beacon_report_interval ?